laurent22/joplin
# Desktop: Fixes #10284: Sort the note list soon after changing a note's title

# Summary

This pull request resolves #10284 by resorting the note list very soon after a note's title is changed from the note editor. 

# Notes

- The current note needs to finish saving before the note list can be resorted. To allow this, `scheduleSaveNote` now returns a `Promise` that resolves after saving has finished.
    - `AsyncActionQueue.waitForAllDone` is used for this. Previously, it used `setInterval`. This pull request should make `waitForAllDone` more efficient.

# Testing plan

1. Open "All notes" in a profile with a large number of notes.
2. Select a note.
3. Switch to alphabetical sorting.
4. Add a `-` to the beginning of the note title.
5. Verify that the notebook list is quickly updated (< 1 second).
6. Remove the `-`.
7. Verify that the note list is quickly updated.

This has  been tested successfully on Ubuntu 24.04.
